erts/preloaded/src/prim_inet.erl

```diff
@@ -2078,21 +2097,58 @@ enc_value_tuple(_, _, _, _) -> [].
 %%
 %% Encoding of option VALUES:
 %%
-enc_value_2(bool, true)     -> [0,0,0,1];
-enc_value_2(bool, false)    -> [0,0,0,0];
-enc_value_2(bool8, true)    -> [1];
-enc_value_2(bool8, false)   -> [0];
-enc_value_2(int, Val)       -> ?int32(Val);
-enc_value_2(uint, Val)      -> ?int32(Val);
-enc_value_2(uint32, Val)    -> ?int32(Val);
-enc_value_2(uint24, Val)    -> ?int24(Val);
-enc_value_2(uint16, Val)    -> ?int16(Val);
-enc_value_2(uint8, Val)     -> ?int8(Val);
-enc_value_2(time, infinity) -> ?int32(-1);
-enc_value_2(time, Val)      -> ?int32(Val);
-enc_value_2(ip,{A,B,C,D})   -> [A,B,C,D];
-enc_value_2(ip, any)        -> ip4_any();
-enc_value_2(ip, loopback)   -> ip4_loopback();
+enc_value_2(bool, true)      -> [0,0,0,1];
+enc_value_2(bool, false)     -> [0,0,0,0];
+enc_value_2(bool8, true)     -> [1];
+enc_value_2(bool8, false)    -> [0];
+enc_value_2(int, Val)        -> ?int32(Val);
+enc_value_2(uint, Val)       -> ?int32(Val);
+enc_value_2(uint32, Val)     -> ?int32(Val);
+enc_value_2(uint24, Val)     -> ?int24(Val);
+enc_value_2(uint16, Val)     -> ?int16(Val);
+enc_value_2(uint8, Val)      -> ?int8(Val);
+enc_value_2(time, infinity)  -> ?int32(-1);
+enc_value_2(time, Val)       -> ?int32(Val);
+%% enc_value_2(ip, IP)
+%%   when (tuple_size(IP) =:= 4) -> ip4_to_bytes(IP);
+%% enc_value_2(ip, any)          -> ip4_any();
+%% enc_value_2(ip, loopback)     -> ip4_loopback();
+enc_value_2(mif, IP)
+  when (tuple_size(IP) =:= 4) -> ip4_to_bytes(IP);
+enc_value_2(mif, Idx)
+  when is_integer(Idx)        -> ?int32(Idx);
+enc_value_2(membership, {IP1, IP2})
+  when (tuple_size(IP1) =:= 4) andalso
+       (tuple_size(IP2) =:= 4) ->
+    enc_value_2(membership, {IP1, IP2, 0});
+enc_value_2(membership, {IP1, IP2})
+  when (tuple_size(IP1) =:= 4) andalso
+       (IP2 =:= any) ->
+    enc_value_2(membership, {IP1, IP2, 0});
+%% enc_value_2(membership, {IP1, any = _IP2})
+%%   when (tuple_size(IP1) =:= 4) ->
+%%     [?INET_AF_INET, ?int32(0), ip4_to_bytes(IP1), ip4_any()];
+enc_value_2(membership, {IP1, IP2, Idx})
+  when (tuple_size(IP1) =:= 4) andalso
+       (tuple_size(IP2) =:= 4) andalso
+       is_integer(Idx) ->
+    %% The reason for turning this thing around (the interface
+    %% before the two address'es) so that we as much as possible
+    %% "look like" IPv6...se below
+    [?int32(?INET_AF_INET), ?int32(Idx), ip4_to_bytes(IP1), ip4_to_bytes(IP2)];
+enc_value_2(membership, {IP1, any = _IP2, Idx})
+  when (tuple_size(IP1) =:= 4) andalso
+       is_integer(Idx) ->
+    [?int32(?INET_AF_INET), ?int32(Idx), ip4_to_bytes(IP1), ip4_any()];
+enc_value_2(membership, {IP, Idx})
+  when (tuple_size(IP) =:= 8) andalso
+       is_integer(Idx) ->
+    %% The reason for turning this thing around (the interface
+    %% before the address) is because of the inet-driver (it reads out a
+    %% 32-bit value for *all* options, so we might as well put a 32-nit
+    %% value "first".
+    [?int32(?INET_AF_INET6), ?int32(Idx), ip6_to_bytes(IP)];
```

Comment:
does this still get encoded into a socket option binary in the same order as using the `raw` option in the workaround described here: https://github.com/erlang/otp/issues/5789#issue-1168721524 ?

Reply:
Sorry for the delay!

Are you referring to: '<<Mad/binary, Ifindex/binary>>'?
Since we use the "proper" type, 'struct ipv6_mreq', I am
not sure why you think the order matter.
Or am I missing something?

Reply:
Oh, didn't realize that. Disregard!